lib/spack/spack/build_systems/sip.py

```diff
@@ -77,12 +55,15 @@ def configure(self, spec, prefix):
 
         args = self.configure_args()
 
+        python_include_dir = 'python' + str(spec['python'].version.up_to(2))
```

Comment:
The `python` package has a `python_include_dir` attribute you should be able to query.

Reply:
Last time I tried either I got the error like "spec doesn't have python attribute" or python include dir was pointing to include dir under python prefix as opposed to pyqtX prefix.

Reply:
You may need to try something like:
```python
inspect.getmodule(self).python_include_dir
```
It _should_ just be `include/python3.7` without any specific prefix, as that's how it's defined in the `python` package.

Reply:
I am getting
```
1 error found in build log:
     4    
  >> 5    configure.py: error: '/mnt/local/spack/opt/spack/linux-centos7-skylake/gcc-7.4.0/py-pyqt5-5.13.1-wh55z5wnc3l35rlfywgij7xdj4ega322/include/python3.7' is not a directory
```
I guess an empty directory needs to be created before the configure step

Reply:
Actually the above error implies that configure is looking for `py-pyqtX_prefix/include/python3.7` as opposed to `py-sip_prefix/include/python3.7` which is incorrect.

Reply:
> You may need to try something like:
> 
> ```python
> inspect.getmodule(self).python_include_dir
> ```
> 
> It _should_ just be `include/python3.7` without any specific prefix, as that's how it's defined in the `python` package. 


The above suggestion yields:
```
     2    #####################################
     3    python_include_dir is
     4    /mnt/local/spack/opt/spack/linux-centos7-skylake/gcc-7.4.0/py-pyqt5-5.13.1-wh55z5wnc3l35rlfywgij7xdj4ega322/include/python3.7
```
not just `include/python3.7` either we can keep the current way of calculating `python_include_dir` or I can do split+join to extract it from `inspect.getmodule(self).python_include_dir`

Reply:
Very interesting. Now I need to figure out how it's adding the prefix. I guess the current way we are doing things is fine then.